mvlearn/decomposition/grouppca.py

```diff
@@ -262,15 +269,42 @@ def transform(self, Xs, y=None):
         """
         check_is_fitted(self)
         Xs = check_Xs(Xs, copy=True)
+
+        if index is None:
+            index_ = np.arange(len(self.individual_projections_))
+        else:
+            index_ = np.copy(index)
+
         if self.multiview_output:
             return [
                 np.dot(X - mean, W.T)
                 for W, X, mean in (
                     zip(
-                        self.individual_projections_, Xs, self.individual_mean_
+                        [self.individual_projections_[i] for i in index_],
+                        Xs,
+                        [self.individual_mean_[i] for i in index_],
                     )
                 )
             ]
+
+        if index is not None:
+            return np.mean(
```

Comment:
For legibility, since the difference between `self.multiview_output` being `True` vs `False` is just the mean, I would compute the list of  `[np.dot(X - mean, W.T) for ....]` outside of the `if self.multiview_output:` and then either return it or return its mean.